Backend: add instructions, target branch, and repo location (#474)
hacs/documentation: source/docs/contribute/backend.md

- How to get started.
- What branch to target.
- How to test locally.

## Finding the backend repository

The backend repository is hosted @ https://github.com/hacs/integration

## Target branch for contributor PRs

> When contributing to the backend, target your PR against the `{{ config.remote_branch }}` branch.

Reviewer: ```suggestion
When contributing to the backend, target your PR against the `main` branch.

```
`config.remote_branch` is for the documentation

## Creating your code changes using the devcontainer

1. Spin up the [devcontainer](/docs/contribute/devcontainer.md).

2. Add your code changes.
3. To start a preview of the application, issue the following command:

    ```bash
    scripts/develop
    ```

    - You have now started a local server that hosts the application on `http://localhost:8123`

> 

Reviewer: ```suggestion

    - To use and see HACS inside the local Home Assistant instance, you need to [configure it in the Home Assistant UI](/docs/use/configuration/basic.md)
```

>     - The server automatically reloads to reflect your changes.

Reviewer: ```suggestion
```

This is currently not possible, and I doubt it ever will be, but if it was a lot of custom integration devs would be happy.

- How to get started.

> - What branch to target.

Reviewer: These 2 items can be removed now.
